amclient/utils.py

```diff
@@ -22,6 +22,38 @@
 METHOD_DELETE = "DELETE"
 
 
+def _call_url(
+    url, params=None, method=METHOD_GET, headers=None, data=None, assume_json=True
+):
+    """Helper to GET a URL.
+
+    :param str url: URL to call
+    :param dict params: Params to pass as HTTP query string
+    :param str method: HTTP method (e.g., 'GET')
+    :param dict headers: HTTP headers
+    :param dict data: Data to pass to request body
+    :param bool assume_json: set to False if the response body should not be
+                             decoded as JSON
+    :returns: Dict of the returned JSON or raises an exception
+    """
+    try:
+        response = requests.request(
+            method, url=url, params=params, headers=headers, data=data
+        )
+        LOGGER.debug("Response: %s", response)
+        LOGGER.debug("type(response.text): %s ", type(response.text))
+        LOGGER.debug("Response content-type: %s", response.headers["content-type"])
+        response.raise_for_status()
+    except Exception:
+        raise
```

Comment:
If the only role of this try/except is to raise the underlying exception and nothing else then I'd say that you don't need the try/except.

Reply:
Good call - I've removed this as well as the try/except in the `if assume_json:` block.

Reply:
Good call - I removed this try/except as well as the one in the `if assume_json` block.